File: maps/src/main/java/it/localhostsoftware/maps/google/GoogleCameraUpdateFactory.java
package it.localhostsoftware.maps.google;

import android.graphics.Point;

import com.google.android.gms.maps.model.CameraPosition;
import com.google.android.gms.maps.model.LatLngBounds;

import it.localhostsoftware.maps.CameraUpdate;
import it.localhostsoftware.maps.CameraUpdateFactory;
import it.localhostsoftware.maps.model.LatLng;


public class GoogleCameraUpdateFactory implements CameraUpdateFactory {
    @Override
    public CameraUpdate<?> zoomIn() {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.zoomIn());
    }

    @Override
    public CameraUpdate<?> zoomOut() {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.zoomOut());
    }

    @Override
    public CameraUpdate<?> scrollBy(float var0, float var1) {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.scrollBy(var0, var1));
    }

    @Override
    public CameraUpdate<?> zoomTo(float var0) {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.zoomTo(var0));
    }

    @Override
    public CameraUpdate<?> zoomBy(float var0) {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.zoomBy(var0));
    }

    @Override
    public CameraUpdate<?> zoomBy(float var0, Point var1) {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.zoomBy(var0, var1));
    }

    @Override
    public CameraUpdate<?> newCameraPosition(CameraPosition var0) {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.newCameraPosition(var0));
    }

    @Override
    public CameraUpdate<?> newLatLng(LatLng<?> var0) {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.newLatLng((com.google.android.gms.maps.model.LatLng) var0.getLatLng()));
    }

    @Override
    public CameraUpdate<?> newLatLngZoom(LatLng<?> var0, float var1) {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.newLatLngZoom((com.google.android.gms.maps.model.LatLng) var0.getLatLng(), var1));
    }

    @Override
    public CameraUpdate<?> newLatLngBounds(LatLngBounds var0, int var1) {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.newLatLngBounds(var0, var1));
    }

    @Override
    public CameraUpdate<?> newLatLngBounds(LatLngBounds var0, int var1, int var2, int var3) {
        return new CameraUpdate<>(com.google.android.gms.maps.CameraUpdateFactory.newLatLngBounds(var0, var1, var2, var3));
    }
}

File: maps/src/main/java/it/localhostsoftware/maps/google/GoogleCameraUpdateFactory.kt
package it.localhostsoftware.maps.google

import android.graphics.Point
import com.google.android.gms.maps.CameraUpdateFactory
import it.localhostsoftware.maps.google.model.GoogleCameraPosition
import it.localhostsoftware.maps.google.model.GoogleLatLng
import it.localhostsoftware.maps.google.model.GoogleLatLngBounds

class GoogleCameraUpdateFactory : it.localhostsoftware.maps.CameraUpdateFactory<GoogleCameraUpdate, GoogleCameraPosition, GoogleLatLng, GoogleLatLngBounds> {
    override fun zoomIn() =
        GoogleCameraUpdate(CameraUpdateFactory.zoomIn())

    override fun zoomOut() =
        GoogleCameraUpdate(CameraUpdateFactory.zoomOut())

    override fun scrollBy(var0: Float, var1: Float) =
        GoogleCameraUpdate(CameraUpdateFactory.scrollBy(var0, var1))

    override fun zoomTo(var0: Float) =
        GoogleCameraUpdate(CameraUpdateFactory.zoomTo(var0))

    override fun zoomBy(var0: Float) =
        GoogleCameraUpdate(CameraUpdateFactory.zoomBy(var0))

    override fun zoomBy(var0: Float, var1: Point) =
        GoogleCameraUpdate(CameraUpdateFactory.zoomBy(var0, var1))

    override fun newCameraPosition(var0: GoogleCameraPosition) =
        GoogleCameraUpdate(CameraUpdateFactory.newCameraPosition(var0.cp))

    override fun newLatLng(var0: GoogleLatLng) =
        GoogleCameraUpdate(CameraUpdateFactory.newLatLng(var0.ll))

    override fun newLatLngZoom(var0: GoogleLatLng, var1: Float) =
        GoogleCameraUpdate(CameraUpdateFactory.newLatLngZoom(var0.ll, var1))

    override fun newLatLngBounds(var0: GoogleLatLngBounds, var1: Int) =
        GoogleCameraUpdate(CameraUpdateFactory.newLatLngBounds(var0.lb, var1))

    override fun newLatLngBounds(var0: GoogleLatLngBounds, var1: Int, var2: Int, var3: Int) =
        GoogleCameraUpdate(CameraUpdateFactory.newLatLngBounds(var0.lb, var1, var2, var3))
}